src/Icicle/Avalanche/FromCore.hs

```diff
@@ -91,7 +91,8 @@ programFromCore namer p
    = factLoop FactLoopNew (C.reduces p)
 
   factLoop loopType reduces
-   = ForeachFacts (namerElemPrefix namer $ namerFact namer) (namerElemPrefix namer $ namerDate namer) (C.input p) loopType
+   = ForeachFacts [ (namerElemPrefix namer $ namerFact namer, C.input p)
+                  , (namerElemPrefix namer $ namerDate namer, DateTimeT) ] (C.input p) loopType
```

Comment:
I think this ``C.input p : ValType`` should be ``T.PairT (C.input p) T.DateTimeT``, so that it covers the whole list of variables